clients/js-legacy/src/actions.ts

```diff
@@ -0,0 +1,272 @@
+ * @param buffer          Record bytes
+ * @param confirmOptions  Options for confirming the transaction
+ *
+ * @return Signature of the confirmed transaction
+ */
+export async function writeRecord(
+    connection: Connection,
+    payer: Signer,
+    record: PublicKey,
+    authority: Signer,
+    offset: number,
+    buffer: Uint8Array,
+    confirmOptions?: ConfirmOptions,
+) {
+    let transactionResults = [];
+    let bufferOffset = 0;
+    let remainingBuffer = buffer;
+    while (bufferOffset < remainingBuffer.length) {
+        const currentChunkBuffer = remainingBuffer.subarray(
+            bufferOffset,
+            bufferOffset + RECORD_CHUNK_SIZE_POST_INITIALIZE
+        );
+        const transaction = new Transaction().add(
+            createWriteInstruction(
+                record,
+                authority.publicKey,
+                offset + bufferOffset,
+                currentChunkBuffer,
+            )
+        );
+        transactionResults.push(sendAndConfirmTransaction(connection, transaction, [payer, authority], confirmOptions));
+        bufferOffset = bufferOffset + RECORD_CHUNK_SIZE_POST_INITIALIZE;
+    }
+    return await Promise.all(transactionResults);
+}
+
+/**
+ * Initialize and write to a record account
+ *
+ * @param connection      Connection to use
+ * @param payer           Payer of the transaction fees
+ * @param record          Record account address
+ * @param authority       Record account authority keypair
+ * @param recordSize      Offset to store the record bytes
+ * @param buffer          Record bytes
+ * @param confirmOptions  Options for confirming the transaction
+ *
+ * @return Signature of the confirmed transaction
+ */
+export async function initializeWriteRecord(
+    connection: Connection,
+    payer: Signer,
+    record: Signer,
+    authority: Signer,
+    offset: number,
+    buffer: Uint8Array,
+    confirmOptions?: ConfirmOptions,
+) {
+    const programId = RECORD_PROGRAM_ID;
+    const recordSize = buffer.length;
+    const recordAccountSize = RECORD_META_DATA_SIZE + recordSize;
+    const lamports = await connection.getMinimumBalanceForRentExemption(recordAccountSize);
+    const firstChunkBuffer = buffer.subarray(0, RECORD_CHUNK_SIZE_PRE_INITIALIZE);
+    const transaction = new Transaction().add(
+        SystemProgram.createAccount({
+            fromPubkey: payer.publicKey,
+            newAccountPubkey: record.publicKey,
+            space: recordAccountSize,
+            lamports,
+            programId,
+        }),
+        createInitializeInstruction(
+            record.publicKey,
+            authority.publicKey,
+        ),
+        createWriteInstruction(
+            record.publicKey,
+            authority.publicKey,
+            offset,
+            firstChunkBuffer,
+        )
+    );
+    await sendAndConfirmTransaction(connection, transaction, [payer, authority, record], confirmOptions);
+
+    if (buffer.length > RECORD_CHUNK_SIZE_PRE_INITIALIZE) {
```

Comment:
if this branch isn't hit, should the function return the initial transaction signature in a list? ie
```
    const signature = await sendAndConfirmTransaction(connection, transaction, [payer, authority, record], confirmOptions);

    if (buffer.length > RECORD_CHUNK_SIZE_PRE_INITIALIZE) {
    ...
    } else {
        return [signature];
    }
```

Reply:
Good point! Fixed in https://github.com/solana-program/record/pull/20/commits/eba95fe7d59c149cfde6f0e55891e0da8ec463e6.